src/screens/Tutorial/TutorialSwiper.js

```diff
@@ -0,0 +1,144 @@
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Button from 'components/Button';
+import Image from 'components/Image';
+
+import { CMS_DATA_TYPES } from 'constants/cmsConstants';
+import { spacing } from 'utils/variables';
+import { getDeviceWidth } from 'utils/common';
+import { images } from 'utils/images';
+
+import TutorialCMSView, { Title, SubTitle } from './TutorialCMSView';
+import TutorialFooter from './TutorialFooter';
+
+type Props = {
+  data: ParsedCMSDocument[],
+  onButtonPress: (val: string) => void,
+  theme: Theme,
+  onFinish: () => void,
+}
+
+const DEVICE_WIDTH = getDeviceWidth();
+const { ONBOARDING_SCREENS_FOR_NATIVES, ONBOARDING_SCREENS_FOR_NEWBIES } = CMS_DATA_TYPES;
+
+const InitialScreenWrapper = styled.View`
+  height: 100%;
+  width: ${DEVICE_WIDTH}px;
+  justify-content: space-between;
+  padding: ${spacing.rhythm}px;
+  align-items: center;
+`;
+
+const ContentWrapper = styled.View`
+  width: 100%;
+  align-items: center;
+`;
+
+const TutorialButton = styled(Button)`
+  margin-bottom: 20px;
+  width: 100%;
+`;
+
+const Img = styled(Image)`
+  margin-top: 40px;
+  width: ${DEVICE_WIDTH * 0.7}px;
+  height: 100px;
+`;
+
+export default withTheme(({
+  onButtonPress, data, onFinish, theme,
+}: Props) => {
+  const scrollViewRef = useRef(null);
+  const [scrollIndex, setScrollIndex] = useState<number>(0);
+  const [userHasScrolled, setUserHasScrolled] = useState<boolean>(false);
+
+  const scrollToIdx = (idx: number) =>
+    scrollViewRef?.current && scrollViewRef.current.scrollTo({ x: idx * DEVICE_WIDTH });
+
+  const handleButtonPress = (val: string) => {
+    onButtonPress(val);
+    scrollToIdx(1);
+  };
+
+  const handleScroll = (e: Object) => {
+    const xPosition = e.nativeEvent.contentOffset.x;
+    if (xPosition === 0) {
+      setScrollIndex(0);
+    } else if (xPosition % DEVICE_WIDTH === 0) {
+      setScrollIndex(xPosition / DEVICE_WIDTH);
+    }
+  };
+
+  const isLast = () => scrollIndex >= data.length;
+
+  const handleNextPress = () => isLast() ? onFinish() : scrollToIdx(scrollIndex + 1);
+
+  const renderInitialScreen = () => (
+    <InitialScreenWrapper>
+      <ContentWrapper>
+        <Title>{t('tutorialTitle')}</Title>
+        <SubTitle>{t('tutorialSubtitle')}</SubTitle>
+        <Img resizeMode="contain" source={images(theme).pillarLogo} />
+      </ContentWrapper>
+      <ContentWrapper>
+        <TutorialButton title={t('button.new')} onPress={() => handleButtonPress(ONBOARDING_SCREENS_FOR_NEWBIES)} />
+        <TutorialButton
+          title={t('button.native')}
+          onPress={() => handleButtonPress(ONBOARDING_SCREENS_FOR_NATIVES)}
+        />
+      </ContentWrapper>
+    </InitialScreenWrapper>
+  );
+
+  return (
+    <ContainerWithHeader headerProps={{ noBack: true, centerItems: [{ title: t('title.welcome') }] }}>
+      <ScrollView
+        horizontal
+        bounces={false}
+        pagingEnabled
+        style={{ width: DEVICE_WIDTH, flex: 1 }}
```

Comment:
Could we try width 100% here?
```suggestion
        style={{ width: `100%`, flex: 1 }}
```